syntax = "proto3";

package calculator;

service Calculator {
  rpc Add (AddRequest) returns (AddResponse);
  rpc CarInfo (DataRequest) returns (stream DataResponse);
}

message AddRequest {
  int32 num1 = 1;
  int32 num2 = 2;
}

message AddResponse {
  int32 result = 1;
}

message DataRequest {}

message DataResponse {
  int32 temp = 1;
  int32 speed = 2;
  int32 rpm = 3;
  int32 volt = 4;
}
